### ISSUE TITLE
Guild Datatext broken again

### ISSUE THREAD

**Issue by waggz81** [Chalsean/XFaction#323]
Recreated all previous troubleshooting steps - 

- close wow
- delete savedvariables/xfaction.lua & .bak
- delete addons/xfaction
- copy fresh copy of 4.1.12 manually to addons
- only loading titan panel, xfaction, debuglog and buggrabber
- modified DT config in xfaction, down to only selecting name and version, not selecting any headers even

Have reports from ElvUI users that they also do not get the roster on mouseover. The DT itself does display in Titan though.

```
Message: Interface/AddOns/XFaction/DataText/DTGuild.lua:112: attempt to index a nil value
Time: Tue Nov 15 19:09:09 2022
Count: 2
Stack: Interface/AddOns/XFaction/DataText/DTGuild.lua:112: attempt to index a nil value
[string "@Interface/AddOns/XFaction/DataText/DTGuild.lua"]:112: in function <Interface/AddOns/XFaction/DataText/DTGuild.lua:96>
[string "@Interface/AddOns/XFaction/DataText/DTGuild.lua"]:292: in function `OnEnter'
[string "@Interface/AddOns/XFaction/DataText/DTGuild.lua"]:29: in function <Interface/AddOns/XFaction/DataText/DTGuild.lua:29>
[string "@Interface/AddOns/Titan/LDBToTitan.lua"]:353: in function <Interface/AddOns/Titan/LDBToTitan.lua:312>

Locals: list = <table> {
}
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 Player-60-0A2BD3D8 = <table> {
 }
 Player-60-0EDC6085 = <table> {
 }
 Player-60-0EDC67AF = <table> {
 }
 Player-60-0EDC78CD = <table> {
 }
 Player-60-0E52F3E9 = <table> {
 }
 Player-60-0EDCC664 = <table> {
 }
 Player-60-0EAFC85D = <table> {
 }
 Player-60-0EDC9B5A = <table> {
 }
 Player-60-0EDC7858 = <table> {
 }
 Player-60-0965FAAE = <table> {
 }
 Player-60-0EDC77CF = <table> {
 }
 Player-60-0EDC44B7 = <table> {
 }
 Player-60-0EDCA36D = <table> {
 }
 Player-60-0EDC4BD9 = <table> {
 }
 Player-60-0EDC7735 = <table> {
 }
 Player-60-0EDCCE7D = <table> {
 }
 Player-60-05753B6A = <table> {
 }
 Player-60-0EB8B307 = <table> {
 }
 Player-60-0EDCE55E = <table> {
 }
 Player-60-0EDCAC1C = <table> {
 }
 Player-60-0EBD9955 = <table> {
 }
 Player-57-0BE42966 = <table> {
 }
 Player-60-06BBBC73 = <table> {
 }
 Player-60-0EDC7008 = <table> {
 }
 Player-60-0EDCE5F8 = <table> {
 }
}
(for control) = "Player-60-0A2BD3D8"
_ = "Player-60-0A2BD3D8"
unit = <table> {
 note = "[ZtH-J] Elexir"
 guild = <table> {
 }
 guid = "Player-60-0A2BD3D8"
 isAlt = true
 class = <table> {
 }
 __name = "Unit"
 profession1 = <table> {
 }
 achievements = 1795
 race = <table> {
 }
 key = "Player-60-0A2BD3D8"
 zone = <table> {
 }
 ID = 256
 level = 60
 mainName = "Elexir"
 realm = <table> {
 }
 initialized = true
 rank = "Veteran Alt"
 itemLevel = 0
 isOnline = true
 timeStamp = 1668560937
 isRunningAddon = false
 guildSpeak = true
 faction = <table> {
 }
 unitName = "Elyxiana-Stormrage"
 isPlayer = false
 name = "Elyxiana"
 factoryKey = "D<jvCgDVaW1"
 profession2 = <table> {
 }
 guildListen = true
 factoryTime = 1668560937
 pvp = ""
}
unitData = <table> {
 GUID = "Player-60-0A2BD3D8"
 Note = "[ZtH-J] Elexir"
 Guild = "Mini Heroes"
 Achievement = 1795
 Race = "Human"
 Name = "Elyxiana"
 ItemLevel = 0
 UnitName = "Elyxiana-Stormrage"
 Level = 60
 Rank = "Veteran Alt"
 Realm = "Stormrage"
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index a nil value"
XFG = <table> {
 Guilds = <table> {
 }
 Channels = <table> {
 }
 Targets = <table> {
 }
 Hooks = <table> {
 }
 Factories = <table> {
 }
 EncodeChatMessage = <function> defined @Interface/AddOns/XFaction/Network/Encode.lua:77
 Specs = <table> {
 }
 Lib = <table> {
 }
 Expansions = <table> {
 }
 Friends = <table> {
 }
 AddonName = "XFaction"
 DeserializeUnitData = <function> defined @Interface/AddOns/XFaction/Network/Decode.lua:50
 Nameplates = <table> {
 }
 SingleLine = <function> defined @Interface/AddOns/XFaction/Developer/Logger.lua:72
 DecodeChatMessage = <function> defined @Interface/AddOns/XFaction/Network/Decode.lua:112
 EncodeBNetMessage = <function> defined @Interface/AddOns/XFaction/Network/Encode.lua:85
 Regions = <table> {
 }
 Races = <table> {
 }
 Handlers = <table> {
 }
 DataDumper = <function> defined @Interface/AddOns/XFaction/Developer/Logger.lua:68
 Options = <table> {
 }
 Versions = <table> {
 }
 Icons = <table> {
 }
 Version = <table> {
 }
 Confederate = <table> {
 }
 Mailbox = <table> {
 }
 Timers = <table> {
 }
 Media = <table> {
 }
 Settings = <table> {
 }
 Realms = <table> {
 }
 Initialized = true
 Links = <table> {
 }
 LoadConfigs = <function> defined @Interface/AddOns/XFaction/Handlers/Addon
 ```

**Comment by waggz81** [comment]
Side note I just noticed when testing for the other issue, when the member count is 0, there's no error displaying the empty frame.

**Comment by Chalsean** [comment]
well one of the bugs finally manifested for me and i fixed it in a new build of 4.1.13.  i dont know if that fixes everything or not but give it a try

**Comment by waggz81** [comment]
It works! huzzah

**Comment by waggz81** [comment]
fixed in [v4.1.14](https://github.com/Chalsean/XFaction/releases/tag/v4.1.14) ([a3d9298](https://github.com/Chalsean/XFaction/commit/a3d9298f9ba29143864b23ebf3e2eb4b1cd52423))
